src/proxy.rs

```diff
@@ -107,17 +109,23 @@ pub fn filter_frontend_request_headers(headers: &Headers) -> Headers {
 ///
 /// The primary purpose of this function is to add and remove headers as required by an
 /// intermediary conforming to the HTTP spec.
-fn map_request(req: server::Request, url: &str) -> client::Request {
-    let mut r = client::Request::new(req.method().clone(), url.parse().unwrap());
-
+fn map_request(req: server::Request, url: Url) -> client::Request {
     let via = create_via_header(
         req.headers().get::<Via>(),
         req.version());
 
     let mut headers = filter_frontend_request_headers(req.headers());
     headers.set(via);
-    r.headers_mut().extend(headers.iter());
 
+    // add host header related to backend
```

Comment:
Yes, this would be required for Amazon S3 as well. I do not think it is the default case, but we need to support it. I have been kicking the _configuration file_ issue down the road as much as I can. Maybe this forces us to deal with it now.